backend/app/main.py

```diff
@@ -366,6 +381,112 @@ async def get_unassigned_geoids(
+    except MultipleResultsFound:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Multiple DistrictrMaps found for Document with ID {document_id}",
+        )
+
+    if districtr_map.child_layer is not None:
+        logger.info(
+            f"Using child layer {districtr_map.child_layer} for document {document_id}"
+        )
+        gerrydb_name = districtr_map.child_layer
+        # TODO: Remove before merging. For temp benching.
+        start_time = time()
+        zone_assignments = contiguity.get_block_assignments(
+            session, document_id, districtr_map.uuid
+        )
+        end_time = time()
+        logger.info(
+            f"Time taken to get block assignments: {end_time - start_time} seconds"
+        )
+    else:
+        gerrydb_name = districtr_map.parent_layer
+        logger.info(
+            f"No child layer configured for document. Defauling to parent layer {gerrydb_name} for document {document_id}"
+        )
+        sql = text("""
+            select zone, array_agg(geo_id) as nodes
+            from assignments
+            where document_id = :document_id group by zone
+            and zone is not null""")
+        result = session.execute(sql, {"document_id": document_id}).fetchall()
+        zone_assignments = [
+            contiguity.ZoneBlockNodes(zone=row.zone, nodes=row.nodes) for row in result
+        ]
+
+    try:
+        path = contiguity.get_gerrydb_graph_file(gerrydb_name)
+    except botocore.exceptions.ClientError as e:
+        # TODO: Maybe in the future this should actually create the graph
+        logger.error(f"Graph not found: {str(e)}")
+        return HTTPException(
+            status_code=404, detail="Graph not found. Unable to compute contiguity"
+        )
+    except Exception as e:
+        return HTTPException(status_code=500, detail=f"Something went wrong: {str(e)}")
+
+    # TODO: Remove before merging. For temp benching.
+    start_time = time()
+    G = await cache.get(gerrydb_name)
+
+    try:
+        if G is None:
+            logger.info(f"Graph not found in cache, loading from {path}")
+            G = contiguity.get_gerrydb_block_graph(path, replace_local_copy=False)
+            assert await cache.set(gerrydb_name, G), "Unable to cache graph"
+        else:
+            logger.info("Graph found in cache")
+    except Exception as e:
+        logger.warning(f"Unable to load and cache graph: {str(e)}")
+
+    end_time = time()
+    logger.info(f"Graph loaded in {end_time - start_time} seconds")
+
+    if not isinstance(G, Graph):
+        raise HTTPException(status_code=500, detail="Error loading graph")
+
+    results = {}
+    for zone_blocks in zone_assignments:
+        logger.info(f"Checking contiguity for zone {zone_blocks.zone}")
+        results[zone_blocks.zone] = contiguity.check_subgraph_contiguity(
+            G=G, subgraph_nodes=zone_blocks.nodes
+        )
+
+    return results
```

Comment:
Q: This works great. What information will help the user fix contiguity? 

Perhaps, returning the extents of contiguous groups somewhat similarly zoom to unassigned?